### PR TITLE
Spiney/bicubic filtering

### PR DESCRIPTION
**Pull request by jdolan**
For review purposes.

### CODE REVIEW

File: src/client/renderer/shaders/default_fs.glsl [hunk 1]
```diff
@@ -74,6 +74,60 @@ vec3 eyeDir;
...
+ * @brief Cubic interpolation helper for textureBicubic().
+ */
+vec4 cubic(float v) {
+	vec4 n = vec4(1.0, 2.0, 3.0, 4.0) - v;
+	vec4 s = n * n * n;
+	float x = s.x;
+	float y = s.y - 4.0 * s.x;
+	float z = s.z - 4.0 * s.y + 6.0 * s.x;
+	float w = 6.0 - x - y - z;
+	return vec4(x, y, z, w) * (1.0/6.0);
```
Comment (jdolan) [inline src/client/renderer/shaders/default_fs.glsl]: Just watch your whitespace in here, please :)

File: src/tools/quemap/lightmap.c [hunk 2]
```diff
@@ -75,8 +75,13 @@ static void BuildLightmapExtents(lightmap_t *lm) {
 		AddStToBounds(st, lm->st_mins, lm->st_maxs);
 	}
 
+	#if 1 // bilinear or bicubic filtering
+	lm->w = floorf(lm->st_maxs[0] - lm->st_mins[0]) + 4;
+	lm->h = floorf(lm->st_maxs[1] - lm->st_mins[1]) + 4;
+	#else // only bilinear filtering
 	lm->w = floorf(lm->st_maxs[0] - lm->st_mins[0]) + 2;
 	lm->h = floorf(lm->st_maxs[1] - lm->st_mins[1]) + 2;
+	#endif
```
Comment (jdolan) [inline src/tools/quemap/lightmap.c]: Feel free to simply change this to the oversampled size that you need, no need for #if.

File: src/client/renderer/shaders/default_fs.glsl (hunk 1 above)
Comment (SpineyPete) [reply src/client/renderer/shaders/default_fs.glsl]: Which parts? (I just totally copy/pasted this code from that source noted in the function below, btw)

Comment (jdolan) [reply src/client/renderer/shaders/default_fs.glsl]: Just `1.0/6.0` should be `1.0 / 6.0` -- nit-picky, I know. I just like to maintain consistent whitespace for the entire codebase.

**Comment by jdolan** [comment]
YEA BUDDAAYYYY

**Pull request merged**